File: datasets/IR-Plag/convertai-original-java-rest-python/case-07/plagiarized/L2/04/multiarray.py
import sys

def main():
    print("Enter a 4 by 4 matrix row by row: ", end="")
    ma = [[float(input()) for _ in range(4)] for _ in range(4)]
    print(f"Sum of the elements in the major diagonal is {sum_diagonal(ma)}")

def sum_diagonal(m):
    return sum(m[i][i] for i in range(len(m)))

if __name__ == "__main__":
    main()

File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/plagiarized/L2/04/multiarray.ts
import * as readline from 'readline';

class multiarray {
  public static main(args: string[]): void {
    const rl = readline.createInterface({
      input: process.stdin,
      output: process.stdout
    });

    console.log('Enter a 4 by 4 matrix row by row:');
    const ma: number[][] = [];

    rl.on('line', (line: string) => {
      const row = line.split(' ').map(Number);
      ma.push(row);
      if (ma.length === 4) {
        rl.close();
        console.log(`Sum of the elements in the major diagonal is ${multiarray.sum(ma)}`);
      }
    });
  }

  public static sum(m: number[][]): number {
    let sum = 0;
    for (let i = 0; i < m.length; i++) {
      sum += m[i][i];
    }
    return sum;
  }
}

multiarray.main(process.argv.slice(2));

